File: crossecore-csharp/Java/Java/Java_/StringLiteralBase.cs
/* CrossEcore is a cross-platform modeling framework that generates C#, TypeScript, 
 * JavaScript, Swift code from Ecore models with embedded OCL (http://www.crossecore.org/).
 * The original Eclipse Modeling Framework is available at https://www.eclipse.org/modeling/emf/.
 * 
 * contributor: Simon Schwichtenberg
 */
 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using oclstdlib;
using Ecore;
namespace Java_{
	public class StringLiteralBase 
	:ExpressionImpl, StringLiteral
	{
		private string _escapedValue = "";
		public virtual string escapedValue
		{
		get { 
			return _escapedValue;
		}
		set { _escapedValue = value; }
		}
		
		protected override EClass eStaticClass() {
			return Java_PackageImpl.Literals.STRINGLITERAL;
		}
		
		
		
		public override Object eGet(int featureID, bool resolve, bool coreType) {
			switch (featureID) {
				case Java_PackageImpl.STRINGLITERAL_COMMENTS:
					return comments;
				case Java_PackageImpl.STRINGLITERAL_ORIGINALCOMPILATIONUNIT:
					return originalCompilationUnit;
				case Java_PackageImpl.STRINGLITERAL_ORIGINALCLASSFILE:
					return originalClassFile;
				case Java_PackageImpl.STRINGLITERAL_ESCAPEDVALUE:
					return escapedValue;
			}
			return base.eGet(featureID, resolve, coreType);
		}
		
		
		public override void eSet(int featureID, object newValue) {
			switch (featureID) {
				case Java_PackageImpl.STRINGLITERAL_COMMENTS:
					comments.Clear();
					comments.AddRange(((List<EObject>)newValue)?.Cast<Comment>());
					return;
				case Java_PackageImpl.STRINGLITERAL_ORIGINALCOMPILATIONUNIT:
					originalCompilationUnit = (CompilationUnit) newValue;
					return;
				case Java_PackageImpl.STRINGLITERAL_ORIGINALCLASSFILE:
					originalClassFile = (ClassFile) newValue;
					return;
				case Java_PackageImpl.STRINGLITERAL_ESCAPEDVALUE:
					escapedValue = (string) newValue;
					return;
			}
			base.eSet(featureID, newValue);
		}
		
		
	}
}

File: crossecore-typescript/java_/StringLiteralBase.ts
import {EClass} from "ecore/EClass";
import {NotificationChain} from "ecore/NotificationChain";
import {ENotificationImpl} from "ecore/ENotificationImpl";
import {NotificationImpl} from "ecore/NotificationImpl";
import {Bag} from "ecore/Bag";
import {InternalEObject} from "ecore/InternalEObject";
import {BasicEObjectImpl} from "ecore/BasicEObjectImpl";
import {Java_PackageLiterals} from "./Java_PackageLiterals";
import {ExpressionImpl} from "./ExpressionImpl";
import {StringLiteral} from "./StringLiteral";
		
			export class StringLiteralBase
			extends ExpressionImpl
			implements StringLiteral
			{
				private _escapedValue:string = '';
				get escapedValue():string{
					return this._escapedValue;
				}
				set escapedValue(value:string){
					this._escapedValue = value; 
				}

			
				public static eStaticClass:EClass;
				
				protected eStaticClass():EClass{
					
					return StringLiteralBase.eStaticClass;
				}
			
			
			
				public eGet_number_boolean_boolean(featureID:number, resolve:boolean, coreType:boolean):any{
					switch (featureID) {
						case Java_PackageLiterals.STRINGLITERAL_COMMENTS:
							return this.comments;
						case Java_PackageLiterals.STRINGLITERAL_ORIGINALCOMPILATIONUNIT:
							return this.originalCompilationUnit;
						case Java_PackageLiterals.STRINGLITERAL_ORIGINALCLASSFILE:
							return this.originalClassFile;
						case Java_PackageLiterals.STRINGLITERAL_ESCAPEDVALUE:
							return this.escapedValue;
					}
					//return this.eGetFromExpression(featureID, resolve, coreType);
					return super.eGet(featureID, resolve, coreType);
				}
				
				public eSet_number_any(featureID:number, newValue:any):void {
					switch (featureID) {
						case Java_PackageLiterals.STRINGLITERAL_ESCAPEDVALUE:
							this.escapedValue = <string> newValue;
							return;
					}
					super.eSet_number_any(featureID, newValue);
				}

				
			}
			
